io/src/test/scala/fs2/io/tcp/SocketSpec.scala

```diff
@@ -127,5 +128,43 @@ class SocketSpec extends Fs2Spec {
           .get
       result shouldBe sizes
     }
+
+    "write - concurrent calls do not cause WritePendingException" in {
+      val message = Chunk.bytes(("123456789012345678901234567890" * 10000).getBytes)
+
+      val localBindAddress =
+        Deferred[IO, InetSocketAddress].unsafeRunSync()
```

Comment:
I think we can just flatMap on it (passing the Deferred as a parameter to `doNothingServer` will also be nice for the nesting problem)